pkg/ui/handlers.go

```diff
@@ -1,37 +1,80 @@
 package ui
 
 import (
+	"fmt"
 	"io/fs"
 	"net/http"
-	"path"
+	"regexp"
+	"strings"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	"github.com/go-chi/chi/v5"
 )
 
 type API struct {
-	fileServer http.Handler
-
-	logger logging.LoggerInterface
+	fileServer   http.Handler
+	fileSystem   fs.FS
+	servePattern *regexp.Regexp
+	htmlRoutes   []string
+	logger       logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	// TODO @shipperizer unsure if we deal with any POST/PUT/PATCH via js
-	mux.HandleFunc("/*", a.uiFiles)
+	mux.HandleFunc("/ui/*", a.uiFiles)
+
+}
+
+func (a *API) getHtmlRoutes() []string {
+	routes := make([]string, 0)
+	nodeList, _ := fs.ReadDir(a.fileSystem, ".")
+
+	for _, node := range nodeList {
+		if match := a.servePattern.FindString(node.Name()); match == ".html" {
+			route := node.Name()[:len(node.Name())-len(match)]
+			routes = append(routes, fmt.Sprintf("/ui/%s", route))
+			continue
+		}
+	}
+	return routes
+}
+
+func (a *API) isValidHtml(route string) bool {
+	for _, h := range a.htmlRoutes {
+		if h == route {
+			return true
+		}
+	}
+	return false
 }
 
 // TODO: Validate response when server error handling is implemented
 func (a *API) uiFiles(w http.ResponseWriter, r *http.Request) {
-	if ext := path.Ext(r.URL.Path); ext == "" && r.URL.Path != "/" {
+
+	if match := a.servePattern.FindString(r.URL.Path); match == ".html" {
+		r.URL.Path = "/ui/404"
+	}
+
+	if a.isValidHtml(r.URL.Path) {
 		r.URL.Path += ".html"
 	}
 
+	url_parts := strings.Split(r.URL.Path, "/")
+	// strip /ui from url path
+	r.URL.Path = fmt.Sprintf("/%s", strings.Join(url_parts[2:], "/"))
+
```

Comment:
i'm a bit lost in what happens here, as we keep changing the `r.URL.Path`

anyway after the consideration by @nsklikas about the `/ui` i guess we could carelessly serve all and just strip the `ui` prefix

not sure if we should add the `.html` suffix as well

Reply:
So what you see here, is that I don't want to serve routes such as /ui/login.html. I'd like to serve routes like /ui/login. At lines 62 to 64, we just take out the /ui part of the url path so that we can serve the requested page or file from the fileserver.